src/ralph/virtual/models.py

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+from django.core.exceptions import ObjectDoesNotExist
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+
+from ralph.assets.models.base import BaseObject
+from ralph.assets.models.choices import ComponentType
+from ralph.assets.models.components import GenericComponent
+from ralph.data_center.models.networks import IPAddress
+from ralph.lib.mixins.models import NamedMixin
+
+
+class CloudProvider(NamedMixin):
+    class Meta:
+        verbose_name = _('Cloud provider')
+        verbose_name_plural = _('Cloud providers')
+
+
+class CloudFlavor(BaseObject):
+    name = models.CharField(_('name'), max_length=255, unique=True)
+    cloudprovider = models.ForeignKey(CloudProvider)
+    flavor_id = models.CharField(unique=True, max_length=100)
+
+    def __str__(self):
+        return format(self.name)
+
+    @property
+    def cores(self):
+        """Number of cores"""
+        try:
+            for component in GenericComponent.objects.filter(asset=self):
```

Comment:
thats exactly why using `self.genericcomponent_set` will be better here - you don't have to worry about field name on the other side :)